### ISSUE TITLE
Can you add a search bar to a collection view controller? 

### ISSUE THREAD

**Issue by eealexander2** [uchicago-mobi/MPCS51030-2019-Winter-Forum#122]
Hey there, is it possible to add a ui search bar to a collection view controller? we're trying to implement a search bar at the top of our screen and then display results in a collection view below. Thanks!

**Comment by chelseatroy** [comment]
Hi Elizabeth!

There is a way to do it: you can set the top navigation bar to a search bar like you did for the What's In the News homework assignment. 

If you want it _in the view_ rather than on the top bar, then you have to architect things just a little differently. Both the `UITableViewController` and `UICollectionViewController` expect that the entire screen will be filled with the table view/collection view in question. 

If you want that view to only be a _part_ of your screen rather than the whole thing, you'll want to make a custom view controller that inherits from `UIViewController.` You can make a scene in storyboard that connects to your custom view controller. Then, you go to the object library, get a collection view, and stick it in the scene of your view controller. You'll control-drag to add an outlet to that view in your custom view controller's code. Then, you can have your ViewController implement a protocol called `UICollectionViewDelegate`. Like `UITableViewDelegate` and `UITableViewDataSouce` do for a table view, this allows you to control a collection view from a ViewController that _isn't_ a `UICollectionViewController`. 

Let me know if you want to do that/if that's not clear and I can help further!

**Comment by eealexander2** [comment]
Ok, thanks! Also, is there a way to have a table view controller segue to different detail view controllers? For example, I have an "Accounts" tab. One cell has account info, etc. and the other two cells contain "Listed Items" and "Rented Items". How do I point them to different Detail View Controllers? Thanks!

**Comment by susanstevens** [comment]
Yes, there is! I would recommend setting up multiple prototype cells in storyboard. Select your table view in storyboard and open the Attributes Inspector. The second option there lets you add more than one prototype cell.

You'll need to give each cell a different identifier and modify your `cellForRow atIndexPath` method. But once you have multiple prototype cells set up, you can add a different segue from each one, pointing to different detail view controllers.

**Comment by eealexander2** [comment]
Great, thank you!

**Comment by eealexander2** [comment]
Another question -  let's say I want to give the user the ability to save an image to an S3 bucket, but I also want to be able to render that saved image when I do a search for the item that the uploaded image is related to. How can I associate the image with the item on the db side? The documentation I'm looking at suggests that it's more efficient to post the image directly to S3 from swift instead of sending it to the server first. Thanks!
